metrics/exporter.go

```diff
@@ -64,17 +65,31 @@ type ExporterOptions struct {
 	// See https://github.com/knative/serving/blob/master/config/config-observability.yaml
 	// for details.
 	ConfigMap map[string]string
+
+	// A lister for Secrets to allow dynamic configuration of outgoing TLS client cert.
+	Secrets clientv1.SecretLister `json:"-"`
```

Comment:
Do we need `json` tag at all?

Reply:
Perhaps we need tags on the others as well, for uniformity?

Reply:
I'm happy to use the default behavior for the others, though I can add tags if you think it's important that we hint that these are JSON-serialized in eventing-contrib (that seems to be the only place where to/from JSON is used at all).